spec/value_status_spec.rb

```diff
@@ -3,62 +3,70 @@
 require 'ronin/recon/worker'
 
 describe Ronin::Recon::ValueStatus do
-  module TestWorkers
-    class TestWorker < Ronin::Recon::Worker
-    end
+  module TestValueStatus
+    class TestWorker < Ronin::Recon::Worker; end
+    class TestWorker2 < Ronin::Recon::Worker; end
   end
 
+  let(:worker_class)  { TestValueStatus::TestWorker }
+  let(:worker_class2) { TestValueStatus::TestWorker2 }
+
   describe "#initialize" do
     it "must initialize #values" do
       expect(subject.values).to eq({})
     end
   end
 
   describe "#value_enqueued" do
-    let(:worker_class)  { TestWorkers::TestWorker }
-    let(:value)         { Ronin::Recon::Values::IP.new('10.12.13.14') }
-    let(:expected_hash) { { foo: :bar, worker_class => :enqueued } }
+    let(:value) { Ronin::Recon::Values::IP.new('192.160.1.1') }
 
     context "when #values has value key" do
+      before do
+        subject.value_enqueued(worker_class2, value)
+      end
+
       it "must add new key worker_class with value :enqueued" do
-        subject.values[value] = { foo: :bar }
-        expect(subject.value_enqueued(worker_class, value)).to eq(:enqueued)
-        expect(subject.values[value]).to eq(expected_hash)
+        subject.value_enqueued(worker_class, value)
+        expect(subject.values[value].size).to eq(2)
+        expect(subject.values[value][worker_class]).to eq(:enqueued)
       end
     end
 
     context "when #values has no value key" do
       it "must create an empty hash and add workerk_class key with value :enqueued" do
         subject.value_enqueued(worker_class, value)
+        expect(subject.values[value].size).to eq(1)
         expect(subject.values[value][worker_class]).to eq(:enqueued)
       end
     end
   end
 
   describe "#job_started" do
-    let(:worker_class)  { TestWorkers::TestWorker }
-    let(:value)         { Ronin::Recon::Values::IP.new('10.12.13.14') }
-    let(:expected_hash) { { foo: :bar, worker_class => :working } }
+    let(:value) { Ronin::Recon::Values::IP.new('192.168.1.1') }
 
     context "when #values has value key" do
+      before do
+        subject.job_started(worker_class2, value)
```

Comment:
To make this more realistic, you could call `subject.value_enqueued(worker_class1, value)` and `subject.value_enqueued(worker_class2, value)`. That way when `subject.job_started(worker_class1, value)` is called, it will transition the status from `:enqueued` to `:working`.